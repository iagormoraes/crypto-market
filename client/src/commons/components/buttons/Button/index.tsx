import { HTMLAttributes } from 'react';
import classnames from 'classnames';
import styles from './style.module.css';

interface Props extends HTMLAttributes<HTMLButtonElement> {
  variant?: ButtonVariant;
  alt?: boolean;
}

export enum ButtonVariant {
  Regular = 'Regular',
  Small = 'Small',
}

export default function Button({
  variant = ButtonVariant.Regular,
  alt,
  children,
  className,
  ...props
}: Props) {
  const variants = {
    [ButtonVariant.Regular]: styles.btnRegular,
    [ButtonVariant.Small]: styles.btnSmall,
  };

  return (
    <button
      className={classnames(
        styles.btn,
        variants[variant],
        { [styles.btnAlt]: alt },
        className,
      )}
      {...props}
    >
      {children}
    </button>
  );
}
