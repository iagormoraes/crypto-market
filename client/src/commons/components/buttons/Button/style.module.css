.btn {
    @apply relative bg-black font-bold text-white shadow-xl transition-all duration-500;
    @apply hover:bg-white hover:text-black;
}

.btn::before,
.btn::after {
    content: '';
    width: 0;
    height: 100%;
    position: absolute;
    inset: 0;

    @apply bg-yellow-300 transition-all duration-500;

    mix-blend-mode: multiply;
}

.btn::after {
    left: unset;
}

.btn:hover::before,
.btn:hover::after {
    width: 50%;
}

.btnRegular {
    @apply px-4 py-1 md:px-8 md:py-2 text-lg md:text-2xl;
}

.btnSmall {
    @apply px-4 py-1 text-xl;
}

.btnAlt {
    @apply bg-white text-black;
}
