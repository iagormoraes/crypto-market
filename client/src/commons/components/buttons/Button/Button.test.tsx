import { render } from '@testing-library/react';

import Button, { ButtonVariant } from '.';

describe('<Button/>', () => {
  it('matches snapshot', () => {
    const { container } = render(<Button>Button</Button>);

    expect(container).toMatchSnapshot();
  });
});
