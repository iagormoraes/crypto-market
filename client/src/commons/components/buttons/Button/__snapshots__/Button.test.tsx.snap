// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> matches snapshot 1`] = `
<div>
  <button
    class="btn btnRegular"
  >
    Button
  </button>
</div>
`;
